lib/trino-parquet/src/main/java/io/trino/parquet/reader/PrimitiveColumnReader.java

```diff
@@ -106,11 +110,31 @@ public static PrimitiveColumnReader createReader(RichColumnDescriptor descriptor
                 return createDecimalColumnReader(descriptor).orElse(new BinaryColumnReader(descriptor));
             case FIXED_LEN_BYTE_ARRAY:
                 return createDecimalColumnReader(descriptor)
-                        .orElseThrow(() -> new TrinoException(NOT_SUPPORTED, " type FIXED_LEN_BYTE_ARRAY supported as DECIMAL; got " + descriptor.getPrimitiveType().getOriginalType()));
+                        .orElseGet(() -> {
+                            // Iceberg 0.11.1 writes UUID as FIXED_LEN_BYTE_ARRAY without logical type annotation (see https://github.com/apache/iceberg/pull/2913)
+                            if (isLogicalUuid(descriptor.getPrimitiveType()) || descriptor.getPrimitiveType().getLogicalTypeAnnotation() == null) {
+                                return new UuidColumnReader(descriptor);
+                            }
+                            throw new TrinoException(NOT_SUPPORTED, " type FIXED_LEN_BYTE_ARRAY supported as DECIMAL; got " + descriptor.getPrimitiveType().getOriginalType());
+                        });
         }
         throw new TrinoException(NOT_SUPPORTED, "Unsupported parquet type: " + descriptor.getPrimitiveType().getPrimitiveTypeName());
     }
 
+    private static boolean isLogicalUuid(PrimitiveType type)
+    {
+        return Optional.ofNullable(type.getLogicalTypeAnnotation())
+                .flatMap(logicalTypeAnnotation -> logicalTypeAnnotation.accept(new LogicalTypeAnnotation.LogicalTypeAnnotationVisitor<Boolean>()
```

Comment:
This can be simplified to
```java
logicalTypeAnnotation.getType() == LogicalTypeToken.UUID

Reply:
yes, except for some reason `LogicalTypeToken` doesn't seem to be public